packages/graphql-transformers-e2e-tests/src/__tests__/NewConnectionTransformer.e2e.test.ts

```diff
@@ -0,0 +1,367 @@
+import { ResourceConstants } from 'graphql-transformer-common'
+import GraphQLTransform from 'graphql-transformer-core'
+import DynamoDBModelTransformer from 'graphql-dynamodb-transformer'
+import KeyTransformer from 'graphql-key-transformer'
+import { ModelConnectionTransformer } from 'graphql-connection-transformer'
+import ModelAuthTransformer from 'graphql-auth-transformer'
+import { CloudFormationClient } from '../CloudFormationClient'
+import { Output } from 'aws-sdk/clients/cloudformation'
+import { GraphQLClient } from '../GraphQLClient'
+import { deploy } from '../deployNestedStacks'
+import emptyBucket from '../emptyBucket';
+import { S3Client } from '../S3Client';
+import * as S3 from 'aws-sdk/clients/s3'
+import * as moment from 'moment';
+
+jest.setTimeout(2000000);
+
+const cf = new CloudFormationClient('us-west-2')
+const customS3Client = new S3Client('us-west-2')
+const awsS3Client = new S3({ region: 'us-west-2' })
+
+const BUILD_TIMESTAMP = moment().format('YYYYMMDDHHmmss')
+const STACK_NAME = `NewConnectionTransformerTest-${BUILD_TIMESTAMP}`
+const BUCKET_NAME = `appsync-new-connection-transformer-test-${BUILD_TIMESTAMP}`
+const LOCAL_FS_BUILD_DIR = '/tmp/new_connection_transform_tests/'
+const S3_ROOT_DIR_KEY = 'deployments'
+
+let GRAPHQL_CLIENT = undefined;
+
+function outputValueSelector(key: string) {
+    return (outputs: Output[]) => {
+        const output = outputs.find((o: Output) => o.OutputKey === key)
+        return output ? output.OutputValue : null
+    }
+}
+
+beforeAll(async () => {
+    const validSchema = `
+type Child
+	@model
+	@key(fields: ["id", "name"])
+{
+	id: ID!
+	name: String!
+
+	parents: [Parent] @connection(keyName: "byChild", fields: ["id"])
+}
+
+type Parent
```

Comment:
It would be great if we could get a reflective `User -> Friendship -> User` scenario covered in the tests since it is an oft-asked for example.

Reply:
Added a test for this.